packages/supermassive/src/__tests__/execute.test.ts

```diff
@@ -295,73 +313,505 @@ query Person($id: Int!) {
     }
     `,
   },
+  // subscription
+  {
+    name: "basic subscription with variables",
+    document: `
+  subscription emitPersons($limit: Int!) {
+    emitPersons(limit: $limit) {
+      name
+      gender
+    }
+  }
+ `,
+    variables: {
+      limit: 5,
+    },
+  },
+  {
+    name: "basic subscription with unused non-required variable",
+    document: `
+  subscription emitPersons($limit: Int!, $throwError: Boolean) {
+    emitPersons(limit: $limit, throwError: $throwError) {
+      name
+      gender
+    }
+  }
+ `,
+    variables: {
+      limit: 5,
+    },
+  },
+  {
+    name: "subscription throw an error",
+    document: `
+  subscription emitPersons($limit: Int!, $throwError: Boolean) {
+    emitPersons(limit: $limit, throwError: $throwError) {
+      name
+      gender
+    }
+  }
+ `,
+    variables: {
+      limit: 5,
+      throwError: true,
+    },
+  },
+  // defer
```

Comment:
There are a few cases where behavior is not quite clear from those tests:

1. Fields that exist inside both - normal selection set, and deferred one (especially nested ones):

```graphql
query {
  person {
    account { signupDate }
    ...PersonDetails @defer
   }
}

fragment PersonDetails on Person {
  account {
    __typename
    signupDate
  }
}
```
(same probably works with streams - especially deeply nested ones)

3. Include/skip inside defer/stream

4. Interplay between defer/stream and include/skip in various combinations, e.g.
```graphql
query($include: Boolean = false) {
  person {
    account { signupDate }
    ...PersonDetails @include(if: $include) @defer
   }
}

fragment PersonDetails on Person {
  account {
    __typename
    signupDate
  }
}
```

5. Error bubbling inside defer/stream

Reply:
5. `defer` for fragment on lists:

```graphql
query {
  someList {
     ...MyListItem @defer
  }
}
fragment MyListItem on ListItem {
  foo
}
```

6. `defer` for fragment on lists of union type:
```graphql
query {
  someListOnUnion {
     ...MyFoo @defer
     ...MyBar
  }
}
fragment MyFoo on UnionMemberFoo {
  foo
}
fragment MyBar on UnionMemberBar {
  bar
}
```

Reply:
Different combinations of fragments on abstract types could be tricky (e.g. fragment on interface not deferred, fragment on specific - deferred and vice-versa